ambry-messageformat/src/main/java/com.github.ambry.messageformat/BlobPropertiesSerDe.java

```diff
@@ -27,43 +27,82 @@ public SystemMetadata(long blobSize, String serviceId, String ownerId, String co
     super(blobSize, serviceId, ownerId, contentType, isPrivate, timeToLiveInSeconds);
     this.creationTimeInMs = creationTime;
   }
+
+  public SystemMetadata(long blobSize, String serviceId, String ownerId, String contentType, boolean isPrivate,
+      long timeToLiveInSeconds, long creationTime, short accountId, short containerId, short issuerAccountId) {
+    super(blobSize, serviceId, ownerId, contentType, isPrivate, timeToLiveInSeconds, accountId, containerId,
+        issuerAccountId);
+    this.creationTimeInMs = creationTime;
+  }
 }
 
 /**
  * Serializes and deserializes BlobProperties
  */
 public class BlobPropertiesSerDe {
 
-  private static final short Version1 = 1;
+  static final short Version1 = 1;
+  static final short Version2 = 2;
   private static final int Version_Field_Size_In_Bytes = 2;
   private static final int TTL_Field_Size_In_Bytes = 8;
   private static final int Private_Field_Size_In_Bytes = 1;
   private static final int CreationTime_Field_Size_In_Bytes = 8;
   private static final int Variable_Field_Size_In_Bytes = 4;
   private static final int BlobSize_Field_Size_In_Bytes = 8;
+  private static final int AccountId_Field_Size_In_Bytes = 2;
+  private static final int ContainerId_Field_Size_In_Bytes = 2;
+  private static final int IssuerAccountId_Field_Size_In_Bytes = 2;
 
-  public static int getBlobPropertiesSize(BlobProperties properties) {
+  // @TODO: remove this to add accountId and containerId once putBlobPropertiesToBuffer is changed to version2
+  public static int getBlobPropertiesSerDeSize(BlobProperties properties) {
     return Version_Field_Size_In_Bytes + TTL_Field_Size_In_Bytes + Private_Field_Size_In_Bytes
         + CreationTime_Field_Size_In_Bytes + BlobSize_Field_Size_In_Bytes + Variable_Field_Size_In_Bytes
         + Utils.getNullableStringLength(properties.getContentType()) + Variable_Field_Size_In_Bytes
         + Utils.getNullableStringLength(properties.getOwnerId()) + Variable_Field_Size_In_Bytes
         + Utils.getNullableStringLength(properties.getServiceId());
   }
 
+  /**
+   * Returns the size of {@link BlobProperties} to serialize in Version 2
+   * @param properties {@link BlobProperties} for which size is requested
+   * @return the size of the {@link BlobProperties} to serialize in Version 2
+   * @TODO: will be used once putBlobPropertiesToBuffer is changed to version2
```

Comment:
Yes, blobProperties is used heavily in lots of places. We have to touch almost 30 to 40 tests as well apart from some prod callers. Would prefer to make the switch during write PR